### PR TITLE
Implements the ADR: #9290 Supplants https://github.com/eclipse/deeplearning4j/pull/9292, adds extensions to presets for discovery by javacpp

### PR DESCRIPTION
**Pull request by agibsonccc**
## What changes were proposed in this pull request?
Implements the ADR: #9290

Adds github workflows for a matrix based build structure for different classifiers.
Also adds proper extensions for presets to ensure javacpp can load the extensions.

## How was this patch tested?
Manually on GH actions

## Quick checklist

The following checklist helps ensure your PR is complete:

- [ X] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [X ] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [X ] Created tests for any significant new code additions.
- [ X] Relevant tests for your changes are passing.

### CODE REVIEW

File: .github/workflows/build-deploy-linux-cuda-11.0.yml [hunk 1]
```diff
@@ -138,33 +158,50 @@ jobs:
...
           cmake --version
           protoc --version
           nvcc --version
           sudo apt-get autoremove
           sudo apt-get clean
-          command="mvn  -Possrh -Dlibnd4j.buildThreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=linux-x86_64 -Dlibnd4j.chip=cuda   --also-make  -Dlibnd4j.helper=cudnn -Pcuda  clean --batch-mode package deploy  -DskipTests"
+<<<<<<< Updated upstream
+          command="mvn  -Possrh -Dlibnd4j.buildThreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=linux-x86_64 -Dlibnd4j.compute=\"5.0 5.2 5.3 6.0 6.2 8.0\" -Dlibnd4j.chip=cuda   --also-make  -Pcuda  clean --batch-mode package deploy  -DskipTests"
+=======
+>>>>>>> Stashed changes
```
Comment (treo) [inline .github/workflows/build-deploy-linux-cuda-11.0.yml]: Merge artifacts.

File: .github/workflows/build-deploy-linux-cuda-11.0.yml [hunk 2]
```diff
@@ -138,33 +158,50 @@ jobs:
...
+               bash ${GITHUB_WORKSPACE}/release-specified-component.sh  "${RELEASE_VERSION}" "${SNAPSHOT_VERSION}" "${RELEASE_REPO_ID}" "${COMMAND}"
               else
-                  echo "Running build and deploying to snapshots"
+<<<<<<< Updated upstream
+                  "Running build and deploying to snapshots"
                   eval "$command"
+=======
+                  echo "Running build and deploying to snapshots with command ${COMMAND}"
+                  eval "${COMMAND}"
+>>>>>>> Stashed changes
```
Comment (treo) [inline .github/workflows/build-deploy-linux-cuda-11.0.yml]: more merge artifacts

File: .github/workflows/build-deploy-linux-cuda-11.2.yml [hunk 3]
```diff
@@ -150,17 +173,24 @@ jobs:
           sudo apt-get autoremove
           sudo apt-get clean
           bash ./change-cuda-versions.sh 11.2
-          command="mvn  -Possrh  -Dlibnd4j.buildThreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=linux-x86_64   -Dlibnd4j.helper=cudnn -Dlibnd4j.chip=cuda clean --batch-mode deploy -DskipTests"
+<<<<<<< Updated upstream
+          command="mvn  -Possrh  -Dlibnd4j.buildThreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=linux-x86_64 -Dlibnd4j.compute=\"5.0 5.2 5.3 6.0 6.2 8.0\"   -Dlibnd4j.chip=cuda clean --batch-mode deploy -DskipTests"
+=======
+>>>>>>> Stashed changes
```
Comment (treo) [inline .github/workflows/build-deploy-linux-cuda-11.2.yml]: And even more

File: .github/workflows/build-deploy-linux-cuda-11.2.yml [hunk 4]
```diff
@@ -150,17 +173,24 @@ jobs:
...
+                    bash ${GITHUB_WORKSPACE}/release-specified-component.sh  "${RELEASE_VERSION}" "${SNAPSHOT_VERSION}" "${RELEASE_REPO_ID}" "${COMMAND}"
                    else
-                       echo "Running build and deploying to snapshots"
+<<<<<<< Updated upstream
+                       "Running build and deploying to snapshots"
                        eval "$command"
+=======
+                       echo "Running build and deploying to snapshots"
+                       eval "${COMMAND}"
+>>>>>>> Stashed changes
```
Comment (treo) [inline .github/workflows/build-deploy-linux-cuda-11.2.yml]: And More

File: .github/workflows/build-deploy-linux-x86_64.yml [hunk 5]
```diff
@@ -54,6 +54,113 @@ on:
 jobs:
   #Note: no -pl here because we publish everything from this branch and use this as the basis for all uploads.
   linux-x86_64:
+    strategy:
+      fail-fast: false
+      matrix:
+<<<<<<< Updated upstream
```
Comment (treo) [inline .github/workflows/build-deploy-linux-x86_64.yml]: More merge artifacts here.

File: .github/workflows/build-deploy-windows-cuda-11.0.yml [hunk 6]
```diff
@@ -153,7 +179,11 @@ jobs:
             ) else (
                       echo "Running snapshots"
                       bash "%GITHUB_WORKSPACE%/bootstrap-libnd4j-from-url.sh"
-                      mvn -Possrh -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64  -Djavacpp.platform=windows-x86_64  %MODULES% --also-make -Dlibnd4j.helper=cudnn -Dlibnd4j.platform=windows-x86_64 -Pcuda -Dlibnd4j.chip=cuda clean  --batch-mode deploy -DskipTests
+<<<<<<< Updated upstream
+                       mvn -Possrh -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64 -Dlibnd4j.compute="5.0 5.2 5.3 6.0 6.2 8.0" -Djavacpp.platform=windows-x86_64  "%MODULES%" --also-make -Dlibnd4j.platform=windows-x86_64 -Pcuda -Dlibnd4j.chip=cuda clean  --batch-mode deploy -DskipTests
+=======
+                      call "%GITHUB_WORKSPACE%\mvn-command.bat"
+>>>>>>> Stashed changes
```
Comment (treo) [inline .github/workflows/build-deploy-windows-cuda-11.0.yml]: some more

File: .github/workflows/build-deploy-windows-cuda-11.2.yml [hunk 7]
```diff
@@ -129,10 +151,17 @@ jobs:
...
           GPG_SIGNING_KEY: ${{ secrets.SONATYPE_GPG_KEY }}
-          COMMAND: "mvn  -Possrh -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64 -Dlibnd4j.compute=\"5.0 5.2 5.3 6.0 6.2 8.0\" -Djavacpp.platform=windows-x86_64 -Dlibnd4j.platform=windows-x86_64 -Pcuda -Dlibnd4j.chip=cuda  -Dlibnd4j.helper=cudnn clean  --batch-mode deploy -DskipTests"
+<<<<<<< Updated upstream
+          COMMAND: "mvn  -Possrh -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64 -Dlibnd4j.compute=\"5.0 5.2 5.3 6.0 6.2 8.0\" -Djavacpp.platform=windows-x86_64 -Dlibnd4j.platform=windows-x86_64 -Pcuda -Dlibnd4j.chip=cuda clean  --batch-mode deploy -DskipTests"
+=======
+          COMMAND: mvn  -Possrh   -Dlibnd4j.helper=${{ matrix.helper }} -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64 -Dlibnd4j.compute=\"5.0 5.2 5.3 6.0 6.2 8.0\" -Djavacpp.platform=windows-x86_64 -Dlibnd4j.platform=windows-x86_64 -Pcuda -Dlibnd4j.chip=cuda  -Dlibnd4j.helper=cudnn clean  --batch-mode deploy -DskipTests
+>>>>>>> Stashed changes
```
Comment (treo) [inline .github/workflows/build-deploy-windows-cuda-11.2.yml]: .

File: .github/workflows/build-deploy-windows-cuda-11.2.yml [hunk 8]
```diff
@@ -153,7 +185,11 @@ jobs:
                     ) else (
                         echo "Running snapshots"
                         bash "%GITHUB_WORKSPACE%/bootstrap-libnd4j-from-url.sh"
-                        mvn -T ${{ github.event.inputs.buildThreads }} -Possrh -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64 -Dlibnd4j.helper=cudnn -Djavacpp.platform=windows-x86_64  -pl ":nd4j-cuda-11.2,:deeplearning4j-cuda-11.2,:libnd4j" --also-make -Dlibnd4j.platform=windows-x86_64 -Pcuda -Dlibnd4j.chip=cuda clean  --batch-mode deploy -DskipTests
+<<<<<<< Updated upstream
+                        mvn -T ${{ github.event.inputs.buildThreads }} -Possrh -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64 -Dlibnd4j.compute="5.0 5.2 5.3 6.0 6.2 8.0" -Djavacpp.platform=windows-x86_64  -pl ":nd4j-cuda-11.2,:deeplearning4j-cuda-11.2,:libnd4j" --also-make -Dlibnd4j.platform=windows-x86_64 -Pcuda -Dlibnd4j.chip=cuda clean  --batch-mode deploy -DskipTests
+=======
+                        call "%GITHUB_WORKSPACE%\mvn-command.bat"
+>>>>>>> Stashed changes
```
Comment (treo) [inline .github/workflows/build-deploy-windows-cuda-11.2.yml]: .

File: .github/workflows/build-deploy-windows.yml [hunk 9]
```diff
@@ -102,23 +107,43 @@ jobs:
...
+                       bash "%GITHUB_WORKSPACE%/release-specified-component.sh"  "%RELEASE_VERSION%" "%SNAPSHOT_VERSION%" "%RELEASE_REPO_ID%" "%COMMAND%"
                  ) else (
                     echo "Running snapshots"
                     bash "%GITHUB_WORKSPACE%/bootstrap-libnd4j-from-url.sh"
-                    mvn  -Possrh -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64  %MODULES% --also-make -Dlibnd4j.platform=windows-x86_64  -Dlibnd4j.chip=cpu  clean --batch-mode deploy -DskipTests
+<<<<<<< Updated upstream
+                    mvn  -Possrh -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64  "%MODULES%" --also-make -Dlibnd4j.platform=windows-x86_64  -Dlibnd4j.chip=cpu  clean --batch-mode deploy -DskipTests
+=======
+                    call "%GITHUB_WORKSPACE%\mvn-command.bat"
+>>>>>>> Stashed changes
```
Comment (treo) [inline .github/workflows/build-deploy-windows.yml]: .

**Review by treo — CHANGES_REQUESTED** [review]
It looks like something went wrong when you were trying to separate out changes, so you've got merge artifacts all over the place. 

I've flagged them as comments so you just need to go through all of them and fix it. But I'd suggest you grep for `<<<<<<<` and see if there is something more left.

File: .github/workflows/build-deploy-linux-x86_64.yml [hunk 10]
```diff
@@ -136,24 +155,32 @@ jobs:
...
+              ls /usr/bin | grep g++
+              sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-7 90
+              sudo update-alternatives --install  /usr/bin/g++ g++ /usr/bin/g++-7  90
+              gcc --version
+          fi
           # NOTE: Complete hack. Find better way later. This moves libgomp.so to a directory where javacpp can find it.
           # For linux, this can be found here: https://github.com/eclipse/deeplearning4j/blob/master/nd4j/nd4j-backends/nd4j-backend-impls/nd4j-native-preset/src/main/java/org/nd4j/nativeblas/Nd4jCpuPresets.java#L150
           # Note also, that the g++ --version as of this writing (May 3,2021) currently returns 5.5.0. This will need to be changed in other versions if there is an update.
-          sudo cp /usr/lib/gcc/x86_64-linux-gnu/`gcc --version | head -n 1 | grep -o '[^ ]*$'` /usr/lib
+          sudo cp "${LIBGOMP_PATH}" /usr/lib
```
Comment (treo) [inline .github/workflows/build-deploy-linux-x86_64.yml:177]: The comment doesn't make sense in this context anymore.

File: .github/workflows/build-deploy-linux-cuda-11.0.yml [hunk 11]
```diff
@@ -122,6 +127,21 @@ jobs:
...
+              elif [ "${HELPER}" != '' ]; then
+                  command="mvn  -Possrh -Dlibnd4j.buildThreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform.extension=-${{ matrix.helper }} -Djavacpp.platform=linux-x86_64 -Dlibnd4j.chip=cuda   --also-make  -Dlibnd4j.helper=${{ matrix.helper }} -Dlibnd4j.extension=${{ matrix.extension }} -Pcuda  clean --batch-mode package deploy  -DskipTests"
+              else
+                  command="mvn  -Possrh -Dlibnd4j.buildThreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform.extension=${{ matrix.extension }} -Djavacpp.platform=linux-x86_64 -Dlibnd4j.chip=cuda   --also-make  -Dlibnd4j.helper=${{ matrix.helper }} -Dlibnd4j.extension=${{ matrix.extension }} -Pcuda  clean --batch-mode package deploy  -DskipTests"
+              fi
+              echo "Setting command for helper ${{ matrix.helper }} and extension ${{ matrix.extension }} to ${command}"
+              echo "COMMAND=${command}" >> $GITHUB_ENV
+
```
Comment (treo) [inline .github/workflows/build-deploy-linux-cuda-11.0.yml]: I'd probably split this some more into a base command and extension parameters. 

Something like
```bash
if [ "${HELPER}" != '' ] && [ "${EXTENSION}" != '' ]; then
    mvn_ext="-Djavacpp.platform.extension=-${{ matrix.helper }}-${{ matrix.extension }} -Dlibnd4j.helper=${{ matrix.helper }} -Dlibnd4j.extension=${{ matrix.extension }}"
elif [ "${HELPER}" != '' ]; then
    mvn_ext="-Djavacpp.platform.extension=-${{ matrix.helper }} -Dlibnd4j.helper=${{ matrix.helper }} -Dlibnd4j.extension=${{ matrix.extension }}"
else
    mvn_ext="-Djavacpp.platform.extension=${{ matrix.extension }} -Dlibnd4j.helper=${{ matrix.helper }} -Dlibnd4j.extension=${{ matrix.extension }}"
fi

command="mvn -Possrh ${mvn_ext} -Dlibnd4j.buildThreads=${{ github.event.inputs.buildThreads }}  -Djavacpp.platform=linux-x86_64 -Dlibnd4j.chip=cuda --also-make -Pcuda clean --batch-mode package deploy -DskipTests"
```

And having refactored it like that, I wonder, why do you even specify those values in the cases where they aren't defined?

**Review by treo — CHANGES_REQUESTED** [review]
Just one more question and a comment.

File: .github/workflows/build-deploy-linux-cuda-11.0.yml (hunk 11 above)
Comment (agibsonccc) [reply .github/workflows/build-deploy-linux-cuda-11.0.yml]: @treo  that was mainly due to environment variable/GH actions debugging. I refactored based on your comments there.

File: .github/workflows/build-deploy-linux-x86_64.yml [hunk 12]
```diff
@@ -112,6 +117,18 @@ jobs:
...
+      - name: Set mvn build command based on matrix
+        shell: bash
+        run: |
+             if [ ${{ matrix.helper }} != '' ] && [ ${{ matrix.extension }} != '' ]; then
+                command="mvn  -X  -Possrh  -Dlibnd4j.helper=${{ matrix.helper }} -Dlibnd4j.extension=${{ matrix.extension }} -Djavacpp.platform.extension=-${{ matrix.helper }}-${{ matrix.extension }} -DskipTestResourceEnforcement=true  -Dmaven.javadoc.failOnError=false -Djavacpp.platform=linux-x86_64 -Dlibnd4j.chip=cpu -Pcpu  --also-make  --batch-mode deploy -DskipTests"
+                elif [ ${{ matrix.helper }} != '' ]; then
+                    command="mvn  -X  -Possrh  -Dlibnd4j.helper=${{ matrix.helper }}  -Djavacpp.platform.extension=-${{ matrix.helper }} -DskipTestResourceEnforcement=true  -Dmaven.javadoc.failOnError=false -Djavacpp.platform=linux-x86_64 -Dlibnd4j.chip=cpu -Pcpu  --also-make  --batch-mode deploy -DskipTests"
+                else
+                    command="mvn  -X  -Possrh  -DskipTestResourceEnforcement=true  -Dmaven.javadoc.failOnError=false -Djavacpp.platform=linux-x86_64 -Dlibnd4j.chip=cpu -Pcpu  --also-make  --batch-mode deploy -DskipTests"
+             fi
```
Comment (treo) [inline .github/workflows/build-deploy-linux-x86_64.yml]: Same kind of refactoring is possible here.

File: .github/workflows/build-deploy-mac.yml [hunk 13]
```diff
@@ -119,6 +124,19 @@ jobs:
...
+      - name: Set mvn build command based on matrix
+        shell: bash
+        run: |
+          if [ ${{ matrix.helper }} != '' ] && [ ${{ matrix.extension }} != '' ]; then
+             command="mvn  -X  -Possrh  -Dlibnd4j.helper=${{ matrix.helper }} -Dlibnd4j.extension=${{ matrix.extension }} -Djavacpp.platform.extension=-${{ matrix.helper }}-${{ matrix.extension }} -DskipTestResourceEnforcement=true  -Dmaven.javadoc.failOnError=false -Djavacpp.platform=macosx-x86_64 -Dlibnd4j.chip=cpu -Pcpu  --also-make  --batch-mode deploy -DskipTests"
+             elif [ ${{ matrix.helper }} != '' ]; then
+                 command="mvn  -X  -Possrh  -Dlibnd4j.helper=${{ matrix.helper }}  -Djavacpp.platform.extension=-${{ matrix.helper }} -DskipTestResourceEnforcement=true  -Dmaven.javadoc.failOnError=false -Djavacpp.platform=macosx-x86_64 -Dlibnd4j.chip=cpu -Pcpu  --also-make  --batch-mode deploy -DskipTests"
+             else
+                 command="mvn  -X  -Possrh  -DskipTestResourceEnforcement=true  -Dmaven.javadoc.failOnError=false -Djavacpp.platform=macosx-x86_64 -Dlibnd4j.chip=cpu -Pcpu  --also-make  --batch-mode deploy -DskipTests"
+          fi
```
Comment (treo) [inline .github/workflows/build-deploy-mac.yml]: and here

File: .github/workflows/build-deploy-windows-cuda-11.0.yml [hunk 14]
```diff
@@ -113,6 +118,22 @@ jobs:
...
+        run: |
+          if ( "${{ matrix.helper }}" -ne "" -And "${{ matrix.extension }}" -ne "" ) {
+              $command="mvn  -Possrh -Dlibnd4j.extension=${{ matrix.extension }} -Djavacpp.platform.extension=-${{ matrix.helper }}-${{ matrix.extension }} -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64  -Dlibnd4j.helper=${{ matrix.helper }}  -Dlibnd4j.platform=windows-x86_64  -Pcuda -Dlibnd4j.chip=cuda  deploy -DskipTests"
+           } elseif ( "${{ matrix.helper }}" -ne "" ) {
+                $command="mvn  -Possrh -Dlibnd4j.extension=${{ matrix.helper }} -Djavacpp.platform.extension=-${{ matrix.helper }} -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64  -Dlibnd4j.helper=${{ matrix.helper }}  -Dlibnd4j.platform=windows-x86_64  -Pcuda -Dlibnd4j.chip=cuda  deploy -DskipTests"
+           } else {
+                  $command="mvn  -Possrh -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64  -Dlibnd4j.helper=${{ matrix.helper }} -Dlibnd4j.platform=windows-x86_64  -Pcuda -Dlibnd4j.chip=cuda  deploy -DskipTests"
+          }
```
Comment (treo) [inline .github/workflows/build-deploy-windows-cuda-11.0.yml]: and here

File: .github/workflows/build-deploy-windows-cuda-11.2.yml [hunk 15]
```diff
@@ -113,6 +118,23 @@ jobs:
...
+        run: |
+          if ( "${{ matrix.helper }}" -ne "" -And "${{ matrix.extension }}" -ne "" ) {
+                 $command="mvn  -Possrh -Dlibnd4j.extension=${{ matrix.extension }} -Djavacpp.platform.extension=${{ matrix.helper }}-${{ matrix.extension }} -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64  -Dlibnd4j.helper=${{ matrix.helper }}  -Pcuda -Dlibnd4j.chip=cuda -Dlibnd4j.platform=windows-x86_64    deploy -DskipTests"
+           } elseif ( "${{ matrix.helper }}" -ne "" ) {
+                $command="mvn  -Possrh -Dlibnd4j.extension=${{ matrix.helper }} -Djavacpp.platform.extension=-${{ matrix.helper }} -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64  -Dlibnd4j.helper=${{ matrix.helper }}  -Pcuda -Dlibnd4j.chip=cuda -Dlibnd4j.platform=windows-x86_64    deploy -DskipTests"
+           } else {
+                  $command="mvn  -Possrh -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64  -Dlibnd4j.helper=${{ matrix.helper }}  -Dlibnd4j.platform=windows-x86_64 -Pcuda -Dlibnd4j.chip=cuda  deploy -DskipTests"
+          }
```
Comment (treo) [inline .github/workflows/build-deploy-windows-cuda-11.2.yml]: and here

File: .github/workflows/build-deploy-windows.yml [hunk 16]
```diff
@@ -102,25 +107,41 @@ jobs:
...
+                if ( "${{ matrix.helper }}" -ne ""  -And "${{ matrix.extension }}" -ne "" ) {
+                      $command="mvn  -Possrh -Dlibnd4j.extension=${{ matrix.extension }} -Djavacpp.platform.extension=-${{ matrix.helper }}-${{ matrix.extension }} -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64  -Dlibnd4j.helper=${{ matrix.helper }}  -Dlibnd4j.platform=windows-x86_64  -Dlibnd4j.chip=cpu  deploy -DskipTests"
+                 } elseif ( "${{ matrix.helper }}" -ne "" ) {
+                      $command="mvn  -Possrh -Dlibnd4j.extension=${{ matrix.extension }} -Djavacpp.platform.extension=-${{ matrix.helper }} -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64  -Dlibnd4j.helper=${{ matrix.helper }} -Dlibnd4j.platform=windows-x86_64  -Dlibnd4j.chip=cpu  deploy -DskipTests"
+                 } else {
+                        $command="mvn  -Possrh -Dlibnd4j.buildthreads=${{ github.event.inputs.buildThreads }} -Djavacpp.platform=windows-x86_64  -Dlibnd4j.helper=${{ matrix.helper }}  -Dlibnd4j.platform=windows-x86_64  -Dlibnd4j.chip=cpu  deploy -DskipTests"
+                }
+
```
Comment (treo) [inline .github/workflows/build-deploy-windows.yml]: and here

**Review by treo — CHANGES_REQUESTED** [review]
A few more places where you can apply that same refactoring. And if ideally, it would be more like the following:
```
if [ "${HELPER}" != '' ] && [ "${EXTENSION}" != '' ]; then
    mvn_ext="-Djavacpp.platform.extension=-${{ matrix.helper }}-${{ matrix.extension }} -Dlibnd4j.helper=${{ matrix.helper }} -Dlibnd4j.extension=${{ matrix.extension }}"
elif [ "${HELPER}" != '' ]; then
    mvn_ext="-Djavacpp.platform.extension=-${{ matrix.helper }} -Dlibnd4j.helper=${{ matrix.helper }}"
else
    mvn_ext=""
fi
```

Or, if you need to have those things set to empty, make it explicit:
```
if [ "${HELPER}" != '' ] && [ "${EXTENSION}" != '' ]; then
    mvn_ext="-Djavacpp.platform.extension=-${{ matrix.helper }}-${{ matrix.extension }} -Dlibnd4j.helper=${{ matrix.helper }} -Dlibnd4j.extension=${{ matrix.extension }}"
elif [ "${HELPER}" != '' ]; then
    mvn_ext="-Djavacpp.platform.extension=-${{ matrix.helper }} -Dlibnd4j.helper=${{ matrix.helper }} -Dlibnd4j.extension="
else
    mvn_ext="-Djavacpp.platform.extension= -Dlibnd4j.helper= -Dlibnd4j.extension="
fi
```

**Comment by agibsonccc** [comment]
@saudet  mainly flagging the presets changes for you.

**Review by saudet — DISMISSED** [review]

**Review by treo — APPROVED** [review]

**Pull request merged**